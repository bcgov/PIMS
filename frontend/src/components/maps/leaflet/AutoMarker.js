import { Marker } from 'react-leaflet';
import React from 'react';

// Create your own class, extending from the Marker class.
const AutoMarker = props => {

  const openPopup = marker => {
    if (marker && props.autoPopup) {
        marker.leafletElement.openPopup();
    }
  };

  return (
    <Marker ref={el => openPopup(el)} {...props}/>
  );
};
export default AutoMarker;