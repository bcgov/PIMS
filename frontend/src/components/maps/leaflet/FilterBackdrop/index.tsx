import * as React from 'react';
import { Spinner } from 'react-bootstrap';
import styled from 'styled-components';

export interface FilterBackdropProps {
  show?: boolean;
}

const Backdrop = styled.div`
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: 999;
  left: 0;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  align-items: center;
  align-content: center;
  justify-items: center;
  justify-content: center;
`;

const FilterBackdrop: React.FC<FilterBackdropProps> = ({ show }) => {
  return show ? (
    <Backdrop>
      <Spinner animation="border" variant="warning" />
    </Backdrop>
  ) : null;
};

export default FilterBackdrop;
