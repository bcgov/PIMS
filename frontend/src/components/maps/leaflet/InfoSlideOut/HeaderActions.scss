@use '../../../../colors';

.link-menu {
  background-color: colors.$filter-background-color;
  margin: 0px -10px;
  font-size: 14px;
  padding: 10px;
  display: flex;
  white-space: nowrap;
  justify-content: flex-start;
  a {
    padding: 0px 2px;
    color: colors.$slide-out-blue;
    display: inline-block;
    white-space: nowrap;
  }
  .vertical-bar {
    border-left: 2px solid colors.$filter-background-text-color;
    height: 18px;
    width: 0;
    padding: 0;
    display: inline-block;
    white-space: nowrap;
  }
  .header-link {
    margin: 0 0.5em;
  }
}
