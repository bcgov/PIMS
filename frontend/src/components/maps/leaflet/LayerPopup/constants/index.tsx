import * as React from 'react';

export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PARCEL_NAME: { label: 'Parcel Name:', display: (data: any) => data.PARCEL_NAME },
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  PLAN_NUMBER: { label: 'Plan Number:', display: (data: any) => data.PLAN_NUMBER },
  PARCEL_CLASS: { label: 'Parcel Class:', display: (data: any) => data.PARCEL_CLASS },
  MUNICIPALITY: { label: 'Municipality:', display: (data: any) => data.MUNICIPALITY },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
  FEATURE_AREA_SQM: {
    label: 'Area:',
    display: (data: any) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_GROUP_NAME: {
    label: 'Administration Area:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
