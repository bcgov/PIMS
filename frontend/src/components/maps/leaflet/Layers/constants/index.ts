export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_NAME: { label: 'Area Name:', display: (data: any) => data.ADMIN_AREA_NAME },
  ADMIN_AREA_ABBREVIATION: {
    label: 'Area Abbrevation:',
    display: (data: any) => data.ADMIN_AREA_ABBREVIATION,
  },
  ADMIN_AREA_GROUP_NAME: {
    label: 'Area Group Name:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
