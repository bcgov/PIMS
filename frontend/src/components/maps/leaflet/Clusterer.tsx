import React, { PropsWithChildren } from 'react';
import { LatLng, LeafletMouseEvent, Icon, DivIcon, Layer, Marker, GeoJSONEvent } from 'leaflet';
import { Map as LeafletMap, TileLayer, Popup, WMSTileLayer, GeoJSON } from 'react-leaflet';
import { BBox, GeoJsonProperties } from 'geojson';
import Supercluster from 'supercluster';
import useSupercluster, { ICluster } from 'hooks/useSupercluster';
import { createSingleMarker, createClusterMarker } from './mapUtils';
import { uniqueId } from 'lodash';

export type ClustererProps<P> = {
  points: Supercluster.PointFeature<P>[];
  bounds?: BBox;
  zoom: number;
  onMarkerClick?: Function;
  onClusterClick?: Function;
};

const Clusterer = <P extends GeoJsonProperties>(props: PropsWithChildren<ClustererProps<P>>) => {
  // state and refs
  const { points, bounds, zoom, onMarkerClick, onClusterClick } = props;

  // get clusters
  // clusters are an array of GeoJSON Feature objects, but some of them
  // represent a cluster of points, and some represent individual points.
  const { clusters, supercluster } = useSupercluster({
    points,
    bounds,
    zoom,
    options: { radius: 75, maxZoom: 18 },
  });

  const pointToLayer = (feature: ICluster, latlng: LatLng): Layer => {
    const { cluster: isCluster } = feature?.properties;
    // we have a cluster to render
    if (!!isCluster) {
      return createClusterMarker(feature, latlng);
    }
    // we have a single point (parcel or building) to render
    return createSingleMarker(feature, latlng);
  };

  const onClick = (e: LeafletMouseEvent) => {
    // the point may be either a cluster or a single map pin
    const cluster = (e?.layer as Marker)?.feature as ICluster;
    const isCluster = cluster?.properties?.cluster;

    // the user clicked on a cluster
    if (isCluster) {
      onClusterClick?.(cluster);
    } else {
      // the user clicked on a single map pin
      onMarkerClick?.(cluster);
    }
  };

  // TODO: improve typing
  return clusters && clusters?.length > 0 ? (
    <GeoJSON
      key={uniqueId('geojson_')}
      data={clusters as any}
      pointToLayer={pointToLayer}
      onclick={onClick}
    ></GeoJSON>
  ) : null;
};

export default Clusterer;
