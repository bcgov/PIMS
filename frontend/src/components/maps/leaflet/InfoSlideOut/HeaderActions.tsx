import variables from '_variables.module.scss';
import { IBuilding, IParcel } from 'actions/parcelsActions';
import { PropertyTypes } from 'constants/propertyTypes';
import queryString from 'query-string';
import * as React from 'react';
import { Row } from 'react-bootstrap';
import { Link, useLocation } from 'react-router-dom';
import styled from 'styled-components';

const LinkMenu = styled(Row)`
  background-color: ${variables.filterBackgroundColor};
  height: 35px;
  width: 322px;
  margin: 0px 0px 5px -10px;
  font-size: 14px;
  padding: 10px;
  a {
    padding: 0px 10px;
    color: ${variables.slideOutBlue};
  }
`;

const VerticalBar = styled.div`
  border-left: 2px solid rgba(96, 96, 96, 0.2);
  height: 18px;
`;

interface IHeaderActions {
  /** The selected property */
  propertyInfo: IParcel | IBuilding | null;
  /** the selected property type */
  propertyTypeId: PropertyTypes | null;
  jumpToView: () => void;
  zoomToView: () => void;
  /** additional action to be taken when a link in the menu is clicked */
  onLinkClick?: () => void;
  /** whether the user has the correct agency/permissions to view all the details */
  canViewDetails: boolean;
  /** whether the user has the correct agency/permissions to edit property details */
  canEditDetails: boolean;
}

/**
 * Actions that can be done on the property info slide out
 * @param propertyInfo the selected property information
 * @param propertyTypeId the selected property type
 * @param onLinkClick additional action on menu item click
 * @param canViewDetails user can view all property details
 * @param canEditDetails user can edit property details
 */
const HeaderActions: React.FC<IHeaderActions> = ({
  propertyInfo,
  propertyTypeId,
  onLinkClick,
  jumpToView,
  zoomToView,
  canViewDetails,
  canEditDetails,
}) => {
  const location = useLocation();

  const buildingId = propertyTypeId === PropertyTypes.BUILDING ? propertyInfo?.id : undefined;
  const parcelId = [PropertyTypes.PARCEL, PropertyTypes.SUBDIVISION].includes(
    propertyTypeId as PropertyTypes,
  )
    ? propertyInfo?.id
    : undefined;
  return (
    <LinkMenu>
      Actions:
      {canViewDetails && (
        <>
          <Link
            onClick={() => {
              jumpToView();
              if (onLinkClick) onLinkClick();
            }}
            to={{
              pathname: `/mapview`,
              search: queryString.stringify({
                ...queryString.parse(location.search),
                sidebar: true,
                disabled: true,
                loadDraft: false,
                buildingId: buildingId,
                parcelId: parcelId,
              }),
            }}
          >
            View details
          </Link>

          {canEditDetails && (
            <>
              <VerticalBar />
              <Link
                onClick={() => {
                  jumpToView();
                  if (onLinkClick) onLinkClick();
                }}
                to={{
                  pathname: `/mapview`,
                  search: queryString.stringify({
                    ...queryString.parse(location.search),
                    disabled: false,
                    sidebar: true,
                    loadDraft: false,
                    buildingId: buildingId,
                    parcelId: parcelId,
                  }),
                }}
              >
                Update
              </Link>
            </>
          )}
          <VerticalBar />
        </>
      )}
      <Link to={{ ...location }} onClick={zoomToView}>
        Zoom map
      </Link>
    </LinkMenu>
  );
};

export default HeaderActions;
