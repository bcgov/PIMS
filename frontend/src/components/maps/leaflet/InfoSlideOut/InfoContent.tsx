import * as React from 'react';
import { ListGroup, Row } from 'react-bootstrap';
import { IBuilding, IParcel } from 'actions/parcelsActions';
import { Label } from 'components/common/Label';
import { ParcelPIDPIN } from './ParcelPIDPIN';
import ParcelAttributes from './ParcelAttributes';
import './InfoSlideOut.scss';
import BuildingAttributes from './BuildingAttributes';
import styled from 'styled-components';
import { ThreeColumnItem } from './ThreeColumnItem';
import variables from '_variables.module.scss';
import { PropertyTypes } from 'constants/propertyTypes';
import useCodeLookups from 'hooks/useLookupCodes';
import { useState } from 'react';
import { ProjectNumberLink } from './ProjectNumberLink';
import { BuildingSvg, LandSvg, SubdivisionSvg } from 'components/common/Icons';

/**
 * Compare two dates to evaluation which is earlier.
 * This should handle 'undefined' values by treating them as earlier.
 * @param a First date to compare.
 * @param b Second date to compare.
 * @returns A negative number indicating ealier, 0 as equal, positive number as later.
 */
export const compareDate = (a: Date | string | undefined, b: Date | string | undefined): number => {
  if (a === undefined && b === undefined) return 0;
  if (a === undefined && b !== undefined) return -1;
  if (a !== undefined && b === undefined) return 1;
  const aDate = typeof a === 'string' ? new Date(a) : (a as Date);
  const bDate = typeof b === 'string' ? new Date(b) : (b as Date);
  return aDate.valueOf() - bDate.valueOf();
};

interface IInfoContent {
  /** the selected property information */
  propertyInfo: IParcel | IBuilding | null;
  /** The property type [Parcel, Building] */
  propertyTypeId: PropertyTypes | null;
  /** whether the user has the correct agency/permissions to view all the details */
  canViewDetails: boolean;
}

export const OuterRow = styled(Row)`
  margin: 0px 0px 10px 0px;
`;

const ContactSres = styled(Row)`
  background-color: #deefff;
  padding: 5px 15px;
  margin: 0px 1px 5px 1px;

  em {
    color: ${variables.darkVariantColor};
  }

  a {
    color: ${variables.slideOutBlue};
  }
`;

const getHeading = (propertyTypeId: PropertyTypes | null) => {
  switch (propertyTypeId) {
    case PropertyTypes.SUBDIVISION:
      return (
        <Label className="header">
          <SubdivisionSvg className="svg" style={{ height: 25, width: 25, marginRight: 5 }} />
          Potential Subdivision
        </Label>
      );
    case PropertyTypes.BUILDING:
      return (
        <Label className="header">
          <BuildingSvg className="svg" style={{ height: 25, width: 25, marginRight: 5 }} />
          Building Identification
        </Label>
      );
    default:
      return (
        <Label className="header">
          <LandSvg className="svg" style={{ height: 25, width: 25, marginRight: 5 }} />
          Parcel Identification
        </Label>
      );
  }
};

/**
 * Component that displays the appropriate information about the selected property
 * in the property info slideout
 * @param {IInfoContent} propertyInfo the selected property
 * @param {IInfoContent} propertyTypeId the property type [Parcel, Building]
 * @param canViewDetails user can view all property details
 */
export const InfoContent: React.FC<IInfoContent> = ({
  propertyInfo,
  propertyTypeId,
  canViewDetails,
}) => {
  const isParcel =
    propertyTypeId !== null &&
    [PropertyTypes.PARCEL, PropertyTypes.SUBDIVISION].includes(propertyTypeId);

  const lookupCodes = useCodeLookups();
  const [privateProject, setPrivateProject] = useState<boolean>(false);

  return (
    <>
      <ListGroup>
        {getHeading(propertyTypeId)}
        {isParcel && <ParcelPIDPIN parcelInfo={propertyInfo as IParcel} />}
        <OuterRow>
          {canViewDetails && (
            <>
              {propertyInfo?.name && (
                <ThreeColumnItem leftSideLabel={'Name'} rightSideItem={propertyInfo?.name} />
              )}
              {propertyInfo?.subAgency ? (
                <>
                  <ThreeColumnItem
                    leftSideLabel={'Ministry'}
                    rightSideItem={lookupCodes.getAgencyFullName(propertyInfo?.agency)}
                  />
                  <ThreeColumnItem
                    leftSideLabel={'Owning agency'}
                    rightSideItem={lookupCodes.getAgencyFullName(propertyInfo.subAgency)}
                  />
                </>
              ) : (
                <ThreeColumnItem
                  leftSideLabel={'Owning ministry'}
                  rightSideItem={lookupCodes.getAgencyFullName(propertyInfo?.agency)}
                />
              )}
            </>
          )}
          <ThreeColumnItem
            leftSideLabel={'Classification'}
            rightSideItem={propertyInfo?.classification}
          />
          {!!propertyInfo?.projectNumbers?.length && (
            <ThreeColumnItem
              leftSideLabel={`Project Number${propertyInfo.projectNumbers.length > 1 ? 's' : ''}`}
              rightSideItem={propertyInfo.projectNumbers.map((projectNum: string) => (
                <ProjectNumberLink
                  key={projectNum}
                  setPrivateProject={setPrivateProject}
                  privateProject={privateProject}
                  agencyId={propertyInfo.agencyId}
                  projectNumber={projectNum}
                  breakLine
                />
              ))}
            />
          )}
        </OuterRow>
        {(!canViewDetails || privateProject) && (
          <ContactSres>
            <em>
              For more information on this property, contact{' '}
              <a href="mailto:RealPropertyDivision.Disposals@gov.bc.ca">
                Strategic Real Estate Services (SRES)
              </a>
            </em>
          </ContactSres>
        )}
      </ListGroup>
      <ListGroup>
        <Label className="header">Location data</Label>
        <OuterRow>
          <ThreeColumnItem
            leftSideLabel={'Civic address'}
            rightSideItem={propertyInfo?.address?.line1}
          />
          <ThreeColumnItem
            leftSideLabel={'Location'}
            rightSideItem={propertyInfo?.address?.administrativeArea}
          />
        </OuterRow>
        <OuterRow>
          <ThreeColumnItem leftSideLabel={'Latitude'} rightSideItem={propertyInfo?.latitude} />
          <ThreeColumnItem leftSideLabel={'Longitude'} rightSideItem={propertyInfo?.longitude} />
        </OuterRow>
      </ListGroup>
      {isParcel && (
        <ParcelAttributes parcelInfo={propertyInfo as IParcel} canViewDetails={canViewDetails} />
      )}
      {propertyTypeId === PropertyTypes.BUILDING && (
        <BuildingAttributes
          buildingInfo={propertyInfo as IBuilding}
          canViewDetails={canViewDetails}
        />
      )}
    </>
  );
};

export default InfoContent;
