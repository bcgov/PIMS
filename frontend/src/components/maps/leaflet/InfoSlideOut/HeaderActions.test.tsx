import * as React from 'react';
import 'leaflet';
import 'leaflet/dist/leaflet.css';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { IParcel, IBuilding } from 'actions/parcelsActions';
import { PropertyTypes } from 'constants/propertyTypes';
import { render } from '@testing-library/react';
import { noop } from 'lodash';
import HeaderActions from './HeaderActions';
import { mockParcel } from './InfoContent.test';

const history = createMemoryHistory();
const mockStore = configureMockStore([thunk]);
const store = mockStore({});

const HeaderComponent = (
  propertyInfo: IParcel | IBuilding | null,
  propertyTypeId: PropertyTypes | null,
  canViewDetails: boolean,
  canEditDetails: boolean,
) => {
  return (
    <Provider store={store}>
      <Router history={history}>
        <HeaderActions
          propertyInfo={propertyInfo}
          propertyTypeId={propertyTypeId}
          canViewDetails={canViewDetails}
          canEditDetails={canEditDetails}
          jumpToView={noop}
          zoomToView={noop}
        />
      </Router>
    </Provider>
  );
};

describe('HeaderActions View', () => {
  it('HeaderActions renders correctly', () => {
    const { container } = render(HeaderComponent(mockParcel, PropertyTypes.PARCEL, true, true));
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Shows all three actions when user has all permissions', () => {
    const { getByText } = render(HeaderComponent(mockParcel, PropertyTypes.PARCEL, true, true));
    expect(getByText('View details')).toBeVisible();
    expect(getByText('Update')).toBeVisible();
    expect(getByText('Zoom map')).toBeVisible();
  });

  it('Shows Zoom and View when user has view permissions', () => {
    const { getByText, queryByText } = render(
      HeaderComponent(mockParcel, PropertyTypes.PARCEL, true, false),
    );
    expect(getByText('View details')).toBeVisible();
    expect(queryByText('Update')).toBeNull();
    expect(getByText('Zoom map')).toBeVisible();
  });

  it('Shows Zoom only when user no permissions', () => {
    const { getByText, queryByText } = render(
      HeaderComponent(mockParcel, PropertyTypes.PARCEL, false, false),
    );
    expect(queryByText('View details')).toBeNull();
    expect(queryByText('Update')).toBeNull();
    expect(getByText('Zoom map')).toBeVisible();
  });
});
