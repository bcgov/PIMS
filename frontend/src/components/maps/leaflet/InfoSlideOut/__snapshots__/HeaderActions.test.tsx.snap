// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderActions View HeaderActions renders correctly 1`] = `
.c0 {
  background-color: filterBackgroundColor;
  height: 35px;
  width: 322px;
  margin: 0px 0px 5px -10px;
  font-size: 14px;
  padding: 10px;
}

.c0 a {
  padding: 0px 10px;
  color: slideOutBlue;
}

.c1 {
  border-left: 2px solid rgba(96,96,96,0.2);
  height: 18px;
}

<div
  class="c0 row"
>
  Actions:
  <a
    href="/mapview?disabled=true&loadDraft=false&parcelId=1&sidebar=true"
  >
    View details
  </a>
  <div
    class="c1"
  />
  <a
    href="/mapview?disabled=false&loadDraft=false&parcelId=1&sidebar=true"
  >
    Update
  </a>
  <div
    class="c1"
  />
  <a
    href="/"
  >
    Zoom map
  </a>
</div>
`;

exports[`InfoContent View InfoContent renders correctly 1`] = `
.c1 {
  margin: 0px;
  width: 300px;
}

.c2 {
  width: 110px;
  max-width: 110px;
  padding-right: 10px;
  padding-left: 0px;
}

.c3 {
  max-width: 1px;
  padding: 0px;
  background-color: rgba(96,96,96,0.2);
}

.c4 {
  padding-left: 10px;
  padding-right: 0px;
}

.c0 {
  margin: 0px 0px 10px 0px;
}

<div
  class="list-group"
>
  <p
    class="label header"
  >
    <svg
      class="svg"
      style="height: 25px; width: 25px; margin-right: 5px;"
    >
      icon-lot.svg
    </svg>
    Parcel Identification
  </p>
  <div
    class="c0 row"
  >
    <div
      class="c1 row"
    >
      <div
        class="c2 col"
      >
        <div
          class="left-side list-group-item"
        >
          <p
            class="label"
          >
            PID
          </p>
        </div>
      </div>
      <div
        class="c3 col"
      />
      <div
        class="c4 col"
      >
        <div
          class="list-group-item"
        >
          000-000-000
        </div>
      </div>
    </div>
    
  </div>
  <div
    class="c0 row"
  >
    <div
      class="c1 row"
    >
      <div
        class="c2 col"
      >
        <div
          class="left-side list-group-item"
        >
          <p
            class="label"
          >
            Name
          </p>
        </div>
      </div>
      <div
        class="c3 col"
      />
      <div
        class="c4 col"
      >
        <div
          class="list-group-item"
        >
          test name
        </div>
      </div>
    </div>
    <div
      class="c1 row"
    >
      <div
        class="c2 col"
      >
        <div
          class="left-side list-group-item"
        >
          <p
            class="label"
          >
            Ministry
          </p>
        </div>
      </div>
      <div
        class="c3 col"
      />
      <div
        class="c4 col"
      >
        <div
          class="list-group-item"
        >
          Ministry of Advanced Education
        </div>
      </div>
    </div>
    <div
      class="c1 row"
    >
      <div
        class="c2 col"
      >
        <div
          class="left-side list-group-item"
        >
          <p
            class="label"
          >
            Owning agency
          </p>
        </div>
      </div>
      <div
        class="c3 col"
      />
      <div
        class="c4 col"
      >
        <div
          class="list-group-item"
        >
          Kwantlen Polytechnic University
        </div>
      </div>
    </div>
    <div
      class="c1 row"
    >
      <div
        class="c2 col"
      >
        <div
          class="left-side list-group-item"
        >
          <p
            class="label"
          >
            Classification
          </p>
        </div>
      </div>
      <div
        class="c3 col"
      />
      <div
        class="c4 col"
      >
        <div
          class="list-group-item"
        >
          Core Operational
        </div>
      </div>
    </div>
  </div>
</div>
`;
